hashring/hashring.go

```diff
@@ -270,20 +276,20 @@ func (r *HashRing) lookupNNoLock(key string, n int) []string {
 	}
 
 	hash := r.hashfunc(key)
-	unique := make(map[string]struct{})
+	unique := make(map[valuetype]struct{})
 
 	// lookup N unique servers from the red-black tree. If we have not
 	// collected all the servers we want, we have reached the
 	// end of the red-black tree and we need to loop around and inspect the
 	// tree starting at 0.
-	r.tree.LookupNUniqueAt(n, hash, unique)
+	r.tree.LookupNUniqueAt(n, replicapoint(hash), unique)
```

Comment:
I am not sure if `hashPoint` will be a better name than `replicaPoint`, when talking about the hashring we always talk about replica points. In one of the later changes the `replicaPoint` gets extended with more data eg. identity and replica number of that identity.

This will look like:
```go
r.tree.LookupNUniqueAt(n, replicaPoint{point:hash}, unique)
```

which is more descriptive then the line above.